app/views/candidate_interface/personal_details/immigration_status/_shared_form.html.erb

```diff
@@ -1,44 +1,44 @@
 <%= f.govuk_error_summary %>
 
-<% if @form.eu_nationality? %>
-  <%= f.govuk_radio_buttons_fieldset(
-    :immigration_status,
-    legend: { text: t('page_titles.immigration_status'), size: 'l' },
-  ) do %>
-    <%= f.govuk_radio_button :immigration_status, 'eu_settled', label: { text: t('application_form.personal_details.immigration_status.values.eu_settled') }, link_errors: true %>
-    <%= f.govuk_radio_button :immigration_status, 'eu_pre_settled', label: { text: t('application_form.personal_details.immigration_status.values.eu_pre_settled') } %>
-    <%= f.govuk_radio_divider 'or' %>
-    <%= f.govuk_radio_button :immigration_status, 'other', label: { text: 'Other' } do %>
-      <%= f.govuk_text_field(:right_to_work_or_study_details, label: { text: 'Enter immigration status' }) %>
+<h1 class="govuk-heading-l"><%= t('page_titles.visa_or_immigration_status') %></h1>
+
+<%= f.govuk_radio_buttons_fieldset :immigration_status,
+  legend: { text: 'Select your visa or immigration status' } do %>
+
+    <% scope = 'application_form.personal_details.immigration_status.values' %>
+
+    <% if @form.eu_nationality? %>
+      <%= f.govuk_radio_button :immigration_status, :eu_settled, label: { text: t('eu_settled', scope:) }, link_errors: true %>
+      <%= f.govuk_radio_button :immigration_status, :eu_pre_settled, label: { text: t('eu_pre_settled', scope:) } %>
     <% end %>
-  <% end %>
-<% else %>
-
-  <h1 class="govuk-heading-l"><%= t('page_titles.visa_or_immigration_status') %></h1>
-  <%= f.govuk_radio_buttons_fieldset :right_to_work_or_study_details,
-    legend: { text: 'Select your visa or immigration status' } do %>
-
-      <% scope = 'application_form.personal_details.immigration_status.values' %>
-      <%= f.govuk_radio_button :immigration_status, :indefinite_leave_to_remain_in_the_uk, label: { text: t('indefinite_leave_to_remain_in_the_uk', scope:) } %>
-      <%= f.govuk_radio_button :immigration_status, :student_visa, label: { text: t('student_visa', scope:) } %>
-      <%= f.govuk_radio_button :immigration_status, :graduate_visa, label: { text: t('graduate_visa', scope:) } %>
-      <%= f.govuk_radio_button :immigration_status, :skilled_worker_visa, label: { text: t('skilled_worker_visa', scope:) } %>
-      <%= f.govuk_radio_button :immigration_status, :dependent_on_partners_or_parents_visa, label: { text: t('dependent_on_partners_or_parents_visa', scope:) } %>
-      <%= f.govuk_radio_button :immigration_status, :family_visa, label: { text: t('family_visa', scope:) } %>
+
```

Comment:
Have we considered moving the list of statuses to a model and iterating over them rather than manually adding them in this partial?

Reply:
Had a chat offline and decided to consider doing this as a refactor as a follow up PR if/when time permits.